src/app/cli/src/coda.ml

```diff
@@ -133,284 +133,312 @@ let daemon logger =
+         in
+         trace_database_initialization "consensus local state" __LOC__
+           trust_dir ;
+         let net_config =
+           { Coda_networking.Config.logger
+           ; trust_system
+           ; time_controller
+           ; consensus_local_state
+           ; gossip_net_params=
+               { timeout= Time.Span.of_sec 3.
+               ; logger
+               ; target_peer_count= 8
+               ; conf_dir
+               ; initial_peers= initial_peers_cleaned
+               ; addrs_and_ports
+               ; trust_system
+               ; max_concurrent_connections } }
+         in
+         let receipt_chain_dir_name = conf_dir ^/ "receipt_chain" in
+         let%bind () = Async.Unix.mkdir ~p:() receipt_chain_dir_name in
+         let receipt_chain_database =
+           Coda_base.Receipt_chain_database.create
+             ~directory:receipt_chain_dir_name
+         in
+         trace_database_initialization "receipt_chain_database" __LOC__
+           receipt_chain_dir_name ;
+         let transaction_database_dir = conf_dir ^/ "transaction" in
+         let%bind () = Async.Unix.mkdir ~p:() transaction_database_dir in
+         let transaction_database =
+           Auxiliary_database.Transaction_database.create ~logger
+             transaction_database_dir
+         in
+         trace_database_initialization "transaction_database" __LOC__
+           transaction_database_dir ;
+         let external_transition_database_dir =
+           conf_dir ^/ "external_transition_database"
+         in
+         let%bind () =
+           Async.Unix.mkdir ~p:() external_transition_database_dir
+         in
+         let external_transition_database =
+           Auxiliary_database.External_transition_database.create ~logger
+             external_transition_database_dir
+         in
+         let%map coda =
+           Coda_lib.create
+             (Coda_lib.Config.make ~logger ~trust_system ~conf_dir ~net_config
+                ~work_selection_method:
+                  (Cli_lib.Arg_type.work_selection_method_to_module
+                     work_selection_method)
+                ?snark_worker_key:run_snark_worker_flag
+                ~snark_pool_disk_location:(conf_dir ^/ "snark_pool")
+                ~wallets_disk_location:(conf_dir ^/ "wallets")
+                ~ledger_db_location:(conf_dir ^/ "ledger_db")
+                ~snark_work_fee:snark_work_fee_flag ~receipt_chain_database
+                ~transition_frontier_location ~time_controller
+                ~initial_propose_keypairs ~monitor ~consensus_local_state
+                ~transaction_database ~external_transition_database ())
+         in
+         { Coda_initialization.coda
+         ; client_whitelist
+         ; rest_server_port
+         ; client_port
+         ; run_snark_worker_action }
        in
-       Coda_run.handle_shutdown ~monitor ~conf_dir coda ;
+       let coda_initialization_deferred = coda_initialization_deferred () in
```

Comment:
This still isn't running inside our monitor.

Reply:
but at least it's captured at the first bind-point (and anecdotally that is sufficient for capturing the exceptions during this period)